MetaMask/metamask-extension
# fix(20853): improve the logic to handle fetch

## **Description**
Fix: https://github.com/MetaMask/metamask-extension/issues/20853
For sentry error: 
https://metamask.sentry.io/issues/3404840685/events/a304e0a6b9784a7780254191cdb0dc59/?project=273505
<img width="1359" alt="Screenshot 2023-09-20 at 21 37 05" src="https://github.com/MetaMask/metamask-extension/assets/12678455/021e6e39-90f4-4ed1-8c4b-c7aedd7d8d5f">

In UI level, the error should be captured and contribute to a state that renders the <add token manually> link. Therefore, we do not need extra UI element to deal with the error. When the error shows up, we want to capture in the `catch` and set that state again just in case.

## **Manual testing steps**
1.  change `{shouldShowNetworkInfo && <NewNetworkInfo />}` to `<NewNetworkInfo />` in `ui/pages/routes/routes.component.js` for testing purpose
2. Enable auto token detection
3. go to network and block the request starting from `0x`
4. You should see the <add link> in the new token information pop up.
 
## **Screenshots/Recordings**

### **Before**

_[screenshot]_

### **After**

[_[screenshot]_](https://github.com/MetaMask/metamask-extension/assets/12678455/c050e9ad-4918-4656-8625-ade2068fbfa7)

## **Related issues**

_Fixes https://github.com/MetaMask/metamask-extension/issues/20853

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
